compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -39,26 +39,44 @@ static LogicalResult applyBufferizeToAllocation(RewriterBase &rewriter,
   return success();
 }
 
-// TODO(avarma): This is a temporary workaround until we have PaddingStrategy
-// This function is to take certain operands from a matmul op and used for
-// new allocation creation. Since we have different pipelines (packing/padding)
-// and padding strategies. We use buffereizeLevel to indicate what pipeline
-// and padding strategy we are targeting.
+/// Utility to get operands from the defining ops of LinalgOp input operands.
+/// For example, we want to promote %pack0 and %pack1 to memory.
+/// %pack0 = tensor.pack % arg0
+/// %pack1 = tensor.pack % arg1
+/// %pack2 = tensor.pack % pack0
+/// %pack3 = tensor.pack % pack1
+/// %generic = linalg.generic ins(%pack2, %pack3)
+static FailureOr<SmallVector<Value>> getOperandsFromDefOp(
+    linalg::LinalgOp &linalgOp) {
+  SmallVector<Value> operands;
+  for (auto input : linalgOp.getDpsInputs()) {
+    auto defOp = input.getDefiningOp();
+    if (!defOp || !defOp->getOperand(0)) {
+      return failure();
+    }
+    operands.push_back(defOp->getOperand(0));
+  }
+  return operands;
+}
+
+// This function helps to fetch operands of either a LinalgOp or its defining
+// ops, based on which operands the caller wants to bufferize via
+// `bufferizeOperand` parameter.
 static FailureOr<SmallVector<Value>> getOperandsToBufferize(
-    int64_t bufferizeLevel, linalg::LinalgOp &linalgOp) {
-  switch (bufferizeLevel) {
-    // Use with packing pipeline, create new allocations for Lhs, Rhs and Out.
-    case -1:
+    BufferizeOperand bufferizeOperand, linalg::LinalgOp &linalgOp) {
+  switch (bufferizeOperand) {
+    // Create new allocations for Lhs, Rhs and Out.
```

Comment:
You may change this to `///` comments following what I wrote in the previous review thread.